src/model/http-request.js

```diff
@@ -0,0 +1,36 @@
+const { Validatable } = require('../mixins/validatable-http-message');
+const clone = require('clone');
+
+class HttpRequest extends Validatable {
+  constructor(resources = {}) {
+    super();
+
+    HttpRequest.resourceKeys.forEach((resourceKey) => {
+      this[resourceKey] = clone(resources[resourceKey], false);
+    });
+  }
+}
+
+HttpRequest.resourceKeys = ['method', 'uri', 'headers', 'body', 'expected'];
```

Comment:
Setting on a class itself, since `this.resourceKeys` fails `gavel-spec` assertion of how classes should look, while `static resourceKeys` is not commonjs syntax.